src/lib.rs

```diff
@@ -175,21 +182,86 @@ enum OfferState {
 }
 
 pub struct OfferHandler {
-    _active_offers: Mutex<HashMap<String, OfferState>>,
+    active_offers: Mutex<HashMap<String, OfferState>>,
     pending_messages: Mutex<Vec<PendingOnionMessage<OffersMessage>>>,
     messenger_utils: MessengerUtilities,
     // This channel will close once the onion messenger has successfully started up.
-    _started: RefCell<Receiver<u32>>,
+    started: RefCell<Receiver<u32>>,
 }
 
 impl OfferHandler {
     pub fn new(started: Receiver<u32>) -> Self {
         OfferHandler {
-            _active_offers: Mutex::new(HashMap::new()),
+            active_offers: Mutex::new(HashMap::new()),
             pending_messages: Mutex::new(Vec::new()),
             messenger_utils: MessengerUtilities::new(),
-            _started: RefCell::new(started),
+            started: RefCell::new(started),
+        }
+    }
+
+    /// Adds an offer to be paid with the amount specified. May only be called once for a single offer.
+    #[allow(clippy::await_holding_refcell_ref)]
+    pub async fn pay_offer(
+        &self,
+        offer: Offer,
+        amount: Option<u64>,
+        network: Network,
+        mut client: Client,
+        blinded_path: BlindedPath,
+        mut reply_path: Option<BlindedPath>,
+    ) -> Result<(), OfferError<Secp256k1Error>> {
+        // Wait for onion messenger to give us the signal that it's ready. Once the onion messenger drops
+        // the channel sender, recv will return None and we'll stop blocking here.
+        while (self.started.borrow_mut().recv().await).is_some() {
+            println!("Error: we shouldn't receive any messages on this channel");
         }
+
+        let validated_amount = validate_amount(&offer, amount).await?;
+
+        // For now we connect directly to the introduction node of the blinded path so we don't need any
+        // intermediate nodes here. In the future we'll query for a full path to the introduction node for
+        // better sender privacy.
+        connect_to_peer(client.clone(), blinded_path.introduction_node_id).await?;
+
+        let offer_id = offer.clone().to_string();
+        {
+            let mut active_offers = self.active_offers.lock().unwrap();
+            if active_offers.contains_key(&offer_id.clone()) {
+                return Err(OfferError::AlreadyProcessing);
+            }
+            active_offers.insert(offer.to_string().clone(), OfferState::OfferAdded);
+        }
+
+        let invoice_request = self
+            .create_invoice_request(client.clone(), offer, vec![], network, validated_amount)
+            .await?;
+
+        if reply_path.is_none() {
```

Comment:
Could you add a comment here? I didn't think that an offer with no reply path was spce-valid?

Reply:
Ah added a comment above to make it clear that this is the reply path we're giving back to the offer creator so it knows where to send back the invoice. I made reply_path an optional parameter of pay_offer mainly just so we could provide a specific reply_path in the integration tests -- maybe it might also be useful later on for end users if they wanted to specify a particular reply_path